query-planner-js/src/buildPlan.ts

```diff
@@ -1118,6 +1123,37 @@ class FetchGroup {
     return rewrites;
   }
 
+  getFindersForInput(inputs?: SelectionSet): Finder[] {
+    // TODO: Cache / optimize
+    assert(this.isEntityFetch, 'Must be a FetchGroup for an entity');
+    const entityName = inputs?.parentType.name;
```

Comment:
That seems to be assuming that the inputs parent type is the entity type, but that's not necessarily the case. `FetchNode` sometimes groups fetch to multiple entity types of the same interface, and in that case the input parent type will be that interface.